<<<<<<< HEAD
---
displayed_sidebar: rulesSidebar
---

:::danger Deprecated

This rule has been moved to the [ESLint stylistic plugin](https://eslint.style).
See [#8072](https://github.com/typescript-eslint/typescript-eslint/issues/8072) and [#8074](https://github.com/typescript-eslint/typescript-eslint/issues/8074) for more information.

:::

<!-- This doc file has been left on purpose to help direct people to the stylistic plugin.

Note that there is no actual way to get to this page in the normal navigation,
so end-users will only be able to get to this page from the search bar. -->
=======
# Enforce consistent indentation (`indent`)

## PLEASE READ THIS ISSUE BEFORE USING THIS RULE [#1824](https://github.com/typescript-eslint/typescript-eslint/issues/1824)

## Rule Details

This rule extends the base [`eslint/indent`](https://eslint.org/docs/rules/indent) rule.
It adds support for TypeScript nodes.

## How to use

```jsonc
{
  // note you must disable the base rule as it can report incorrect errors
  "indent": "off",
  "@typescript-eslint/indent": ["error"]
}
```

## Options

See [`eslint/indent` options](https://eslint.org/docs/rules/indent#options).

<sup>Taken with ❤️ [from ESLint core](https://github.com/eslint/eslint/blob/master/docs/rules/indent.md)</sup>
>>>>>>> packages
